ci-operator/step-registry/vsphere/provision/bastionhost/vsphere-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+bastion_name="${CLUSTER_NAME}-bastion"
+bastion_ignition_file="${SHARED_DIR}/${CLUSTER_NAME}-bastion.ign"
+if [[ ! -f "${bastion_ignition_file}" ]]; then
+  echo "'${bastion_ignition_file}' not found, abort." && exit 1
+fi
+bastion_ignition_base64=$(base64 -w0 < "${bastion_ignition_file}")
+
+echo "$(date -u --rfc-3339=seconds) - Configuring govc exports..."
+# shellcheck source=/dev/null
+source "${SHARED_DIR}/govc.sh"
+
+#Get available ova template
+echo "$(date -u --rfc-3339=seconds) - Get avaiable ova template..."
+vm_template="${BASTION_OVA_URI##*/}"
+
+if [[ "$(govc vm.info ${vm_template} | wc -c)" -eq 0 ]] && [[ "$(govc vm.info ${vm_template}-bastion | wc -c)" -eq 0 ]]
+then
+  echo "${vm_template} and ${vm_template}-bastion does not exist, creating it from ${BASTION_OVA_URI}..."
+
+  cat > /tmp/rhcos.json << EOF
+{
+   "DiskProvisioning": "thin",
+   "MarkAsTemplate": false,
+   "PowerOn": false,
+   "InjectOvfEnv": false,
+   "WaitForIP": false,
+   "Name": "${vm_template}-bastion",
+   "NetworkMapping":[{"Name":"VM Network","Network":"${BASTION_HOST_SUBNET}"}]
+}
+EOF
+
+  curl -L -o /tmp/rhcos.ova "${BASTION_OVA_URI}"
+  govc import.ova -options=/tmp/rhcos.json /tmp/rhcos.ova &
+  vm_template="${vm_template}-bastion"
+  wait "$!"
+fi
+
+echo "ova template: ${vm_template}"
```

Comment:
If `${vm_template}-bastion` is already existing, no need to import, seem like the logic here is not right?

Reply:
If both `${vm_template}` and `${vm_template}-bastion` does not exist, then go into `if` block, upload template with name `${vm_template}-bastion`, and set vm_template as ${vm_template}-bastion , seems no issue?

Reply:
Cluster-1 is being installed, both ${vm_template} and ${vm_template}-bastion does not exist, upload template with name ${vm_template}-bastion, running the installation. After that, Cluster-2 installation triggered, `vm_template` will be set to `rhcos-410.84.202207061638-0-vmware.x86_64.ova` firstly, come to the `if` block, `${vm_template}-bastion` is existing, exit the `if` block, come to this line, `vm_template` is still pointed to `rhcos-410.84.202207061638-0-vmware.x86_64.ova`, while the vm_template does not exist, only `${vm_template}-bastion` exist, right?

Reply:
yes, you are right, thanks for detailed explanation, updated.